app/controllers/oral_history_access_requests_controller.rb

```diff
@@ -0,0 +1,43 @@
+# PUBLIC FACING
+# Staff-facing actions are in app/controllers/admin/oral_history_access_requests_controller.rb
+class OralHistoryAccessRequestsController < ApplicationController
+
+  # GET /works/4j03d097t/request_oral_history_access
+  def new
+    @work_friendlier_id = params['work_friendlier_id']
+    set_work
+    @oral_history_access_request = Admin::OralHistoryAccessRequest.new
+  end
+
+  # POST "/request_oral_history_access"
+  def create
+    @work_friendlier_id = params['admin_oral_history_access_request'].delete('work_friendlier_id')
+    set_work
+    @oral_history_access_request = Admin::OralHistoryAccessRequest.new(oral_history_access_request_params)
+    @oral_history_access_request.work = @work
+    if @oral_history_access_request.save
+
+      # redirect_to work_path(@work.friendlier_id), notice: 'Your request has been logged.'
+      render plain: "This functionality is not activated yet, and is only present for testing."
+    else
+     render :new
+    end
+  end
+
+private
+  def oral_history_access_request_params
+    params.require(:admin_oral_history_access_request).permit(
+      :work_friendlier_id, :patron_name, :patron_email,
+      :patron_institution, :intended_use, :status, :notes)
+  end
+
+  def set_work
```

Comment:
How about `set_work(friendlier_id)` instead of relying on instance variable `@friendlier_id` as state? 

Or even how about `@work = load_work(params['work_friendlier_id'])` for even more explicitness? Now `load_work` neither relies on nor sets mysterious instance state, it's just a function with inputs and outputs all visible. 

I know Rails really likes instance variables, and they are the standard way of communicating with views. But they are also rightly widely critisized for making the coupling between code really mysterious. When `@work_friendlier_id` is not being used in the view, there is no reason not to make this code much more explicit and standard good design.

Reply:
That works!